module-system/sov-modules-macros/README.md

```diff
@@ -0,0 +1,36 @@
+# sov-modules-macros
+
+This crate provides Rust macros specifically designed to be used with the Sovereign `module-system`. When developing a module, the  developer's primary focus is on implementing the business logic, without having to worry about low-level details such as message serialization/deserialization or how messages are dispatched to the appropriate module.
+
+To alleviate the burden of writing repetitive and mechanical code, this crate offers a collection of macros that generate the necessary boilerplate code. 
+
+The following derive macros are supported:
+
+1. The `ModuleInfo`: Derives the `sov-modules-api::ModuleInfo` implementation for the underlying type. 
+1. The `Genesis`: Derives the `sov-modules-api::Genesis` implementation for the underlying type.
+1. The `DispatchCall`: Derives the `sov-modules-api::DispatchCall` implementation for the underlying type.
+1. The `DispatchQuery`: Derives the `sov-modules-api::DispatchQuery` implementation for the underlying type.
+1. The `MessageCodec`: Adds message serialization/deserialization functionality to the underlying type.
```

Comment:
nit: Can we add a link to the documentation for these traits?

Reply:
done